provider/k8s/instance_test.go

```diff
@@ -35,3 +49,61 @@ func nodeCreateResources(c kubernetes.Interface, name, cpu, mem string) error {
 		}
 	})
 }
+
+func TestInstanceShell(t *testing.T) {
```

Comment:
I'd rename this `TestInstanceShellError`  since you are asserting the `InstanceShell()` call for errors(I'm guessing bc we can't mock the `ssh.Dial` call.
For the "happy path" case I think it's worth adding an e2e test with keyroll and ssh if possible

Reply:
Changed the name